api/v1alpha1/provider_spec.go

```diff
@@ -12,7 +12,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
```

Comment:
Should this location be `api/v1alpha1/provider_spec.go ` or `pkg/api/v1alpha1/provider_spec.go`. I would think the latter?

Reply:
Hi @prashanth26,

[Here](https://github.com/gardener/etcd-druid/pull/169) is a reference PR I referred to while I created this dedicated module of `api`. The dedicated module should ideally be in the root is what I believe to avoid the cyclic dependency issues. Let me know your thoughts.

Reply:
Hi @AxiomSamarth ,

My first impression is that also long as they are in different packages it's shouldn't matter. However, I am not the expert here. @amshuman-kr knows better and probably has the right opinion here?

Reply:
Yes, I have updated the location of the api module to `/pkg/api/v1alpha1`